docs/developer/watcher-local-test.md

```diff
@@ -0,0 +1,136 @@
+export KLM_PATH=<local-path-of-kyma-lifecycle-manager-project>
+export KLM_VERSION=<docker-image-tag-for-local-testing>
+export KUBE_CONFIG_DIR=<path-to-export-local-kubeconfigs-to>
+```
+
+### Create KCP cluster
+Run the following command to create a local control-plane (KCP) cluster:
+```shell
+k3d cluster create kcp-local --port 9080:80@loadbalancer \
+--registry-create k3d-registry.localhost:0.0.0.0:5111 \
+--k3s-arg '--no-deploy=traefik@server:0'
+```
+
+### Install Istio and Manifest CRD
+Run the following command to install Istio and Manifest CRD on the KCP cluster:
+```shell
+istioctl install -y && \
+kubectl apply -f https://raw.githubusercontent.com/kyma-project/module-manager/main/config/crd/bases/operator.kyma-project.io_manifests.yaml
+```
+
+### Build and push lifecycle-manager image
+Run the following commands to build and push lifecycle-manager image to local k3d registry:
+```shell
+kubectl config use-context k3d-kcp-local && \
+export K3D_REG=k3d-registry.localhost:5111 && \
+make -C $KLM_PATH docker-build IMG=$K3D_REG/lifecycle-manager:$KLM_VERSION && \
+make -C $KLM_PATH docker-push IMG=$K3D_REG/lifecycle-manager:$KLM_VERSION
+```
+
+### Deploy lifecycle-manager on KCP:
+Run the following command to deploy lifecycle-manager on the KCP cluster:
+```shell
+kubectl config use-context k3d-kcp-local && \
+export K3D_REG=k3d-registry.localhost:5111 && \
+make -C $KLM_PATH local-deploy-with-watcher IMG=$K3D_REG/lifecycle-manager:$KLM_VERSION
+```
+
+### Apply sample module templates for sample-kyma:
+Run the following command to apply sample module templates needed for sample Kyma on the KCP cluster:
+```shell
+kubectl config use-context k3d-kcp-local && \
+kubectl apply -f $KLM_PATH/config/samples/component-integration-installed/operator_v1alpha1_moduletemplate_kcp-module.yaml && \
+kubectl apply -f $KLM_PATH/config/samples/component-integration-installed/operator_v1alpha1_moduletemplate_skr-module.yaml
+```
+### Apply sample Kyma CR
+1. Run the following command to create a local kyma-runtime (SKR) cluster:
+```shell
+k3d cluster create skr-local
+```
+2. Run the following command to generate and apply sample Kyma CR and its corresponding secret on KCP:
+```shell
+kubectl config use-context k3d-kcp-local && \
+cat << EOF | kubectl apply -f -
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kyma-sample
+  namespace: default
+  labels:
+    "operator.kyma-project.io/kyma-name": "kyma-sample"
+    "operator.kyma-project.io/managed-by": "lifecycle-manager"
+data:
+  config: $(echo "${$(k3d kubeconfig get skr-local)/0.0.0.0/host.k3d.internal}" | base64)
+---
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: Kyma
+metadata:
+  name: kyma-sample
+  namespace: default
+spec:
+  channel: regular
+  sync:
+    enabled: true
+  modules:
+    - name: kcp-module
+EOF
+```
+### Verify that watcher-webhook is installed on SKR
+By checking the `Kyma CR` events, verify that the `SKRWebhookIsReady` ready condition is set to `True`.
```

Comment:
For me this step is never reached. Instead I face this error:

```
│ {"level":"ERROR","date":"2022-12-23T10:30:38.70169913Z","caller":"controller/controller.go:326","msg":"Reconciler error","context":{"controller":"kyma","controllerGroup":"operator.kyma-project.io","c │
│ ontrollerKind":"Kyma","kyma":{"name":"kyma-sample","namespace":"default"},"namespace":"default","name":"kyma-sample","reconcileID":"485ec836-31eb-4cb0-8ac4-7e560cdcf136","error":"failed to verify web │
│ hook resources: unable to create manifest processor: open /skr-webhook/manifest: permission denied"}
```

Thus both kyma CRs are in error state.

Reply:
Also the SKR Condition is not set to false but simply does not exist so I think the Condition is actually not set correctly in the loop.

Reply:
I have tried to reproduce it but didn't succeed. It would be nice if you could try it again with the latest and verify if the same error is happening again?